pkg/kube/object_patch/patch.go

```diff
@@ -55,6 +65,39 @@ func ParseSpecs(specBytes []byte) ([]OperationSpec, error) {
 	return specs, validationErrors.ErrorOrNil()
 }
 
+func (o *ObjectPatcher) fakeClusterPreferredVersion(spec OperationSpec) (OperationSpec, error) {
```

Comment:
The name of the function is misleading since it also applies to the real client. `clusterPreferredVersion`?

Reply:
Well, it's not applied to the real client because of
```
if len(preferred) > 0 {
	// its not a fake cluster, we can hold empty version, it works for production
	return spec, nil
}
```

inside the function

Reply:
Nice.